src/wmtk/PolygonMesh.cpp

```diff
@@ -0,0 +1,239 @@
+#include "PolygonMesh.hpp"
+
+namespace wmtk {
+PolygonMesh::PolygonMesh()
+    : Mesh(2)
+    , m_en_handle(register_attribute<long>("m_en", PrimitiveType::Edge, 2))
```

Comment:
Ideally we would have both Edges and HalfEdges (as some attributes like edge length would be ideally represented on just an edge. Perhaps we should add HalfEdge as a PrimitiveType (or rename it) this sort of change should be added to the main repo as its own pull request.

Reply:
The `PrimitiveType` enum is used for other places, such as in the `switch_tuple` method. I'm somewhat concerned that adding a halfedge type will give ambiguous meaning and undefined behavior for functions that assume a primitive type is an n-dimensional cell.

On the other hand, halfedge attributes seem generally useful to me, such as for per-corner data like uv positions. Also, a general polyhedral mesh could not be represented using vertex, edge, face, or tet attributes, but it could be implemented using halfedge attributes.

The difference between an edge attribute and a halfedge attribute based implementation of a polygonal mesh are very minor, so I don't have a strong preference and can change which is used later without much effort. However, it would be good to have a settled answer on which we actually want. I can add the pull request to the main repo to continue the conversation there.